library/openssl/kdf/pbkdf2_hmac_spec.rb

```diff
@@ -0,0 +1,283 @@
+require_relative '../../../spec_helper'
+require 'openssl'
+
+describe "OpenSSL::KDF.pbkdf2_hmac" do
+  it "creates the same value with the same input" do
+    pass = "secret"
+    salt = "\x00".b * 16
+    iterations = 20_000
+    length = 16
+    hash = "sha1"
+    key = OpenSSL::KDF.pbkdf2_hmac(pass, salt: salt, iterations: iterations, length: length, hash: hash)
+    key.should == "!\x99+\xF0^\xD0\x8BM\x158\xC4\xAC\x9C\xF1\xF0\xE0".b
```

Comment:
Yeah, I wasn't exactly proud of this repetitive code either.

Since it's mostly just a bunch of default with one argument changed or removed, I was thinking maybe it's better to create a set of defaults, and the tests should explicitly alter these values. Something like this:
```ruby
before :each do
  @pass = "secret"
  @defaults = { salt: "\x00".b * 16, iterations: 20_000, ... }
end

it "works with 10_000 iterations do"
  key = OpenSSL::KDF.pbkdf2_hmac(@pass, **@defaults, iterations: 10_000)
  key.should == SOME_VALUE
end
```

This reduces the length of the tests, removes a lot of duplication, and makes it a bit more clear which parameter we're testing.

Reply:
I pushed an update with the defaults hash, I would think this is much clearer and concise.